packages/editor/src/lib/editor/Editor.ts

```diff
@@ -8271,21 +8271,69 @@ export class Editor extends EventEmitter<TLEventMap> {
 	}
 
 	/**
-	 * Dispatch a focus event.
+	 * Puts the editor into focused mode.
+	 *
+	 * This makes the editor eligible to receive keyboard events and some pointer events (move, wheel).
 	 *
 	 * @example
 	 * ```ts
 	 * editor.focus()
 	 * ```
 	 *
+	 * By default this also dispatches a 'focus' event to the container element. To prevent this, pass `focusContainer: false`.
+	 *
+	 * @example
+	 * ```ts
+	 * editor.focus({ focusContainer: false })
+	 * ```
+	 *
+	 * @public
+	 */
+	focus({ focusContainer = true } = {}): this {
+		if (focusContainer) {
```

Comment:
Yeah it's niche but you might be calling `.focus` and `.blur` from code paths that have already essentially handled the 'focus' and 'blur' logic, and where dispatching events may trigger other event handlers higher in the tree that you might not want to be triggered because it might duplicate or trample work you've already done.

Reply:
i guessss but i can't think of any other SDK/API that works like this. feels like a footgun in that i might get into a half-unfocused state 🤔